### PR TITLE
macos soundviz

### PR DESCRIPTION
**Pull request by zomfg**
So I had a go with this, it detects the right sound frequencies and the result looks similar to the Windows version.
There is a limitation: by default it seems that you can't (reasonably easily) record desktop sound, you only have access to inputs (mic, line-in) and need a third party software to re-route desktop sound to virtual input (I put that in a help dialog). I guess better than nothing...

Notable changes:
 * renamed `BASS_SOUND_SUPPORT` to `SOUNDVIZ_SUPPORT` in source code, in project files it's defined by default for macOS, and for Windows it's defined if bass define is there, so `build-vars` file works same as before
 * refactored `SoundManager` and moved platform specific stuff in daughter classes
 * changed sound timers to precise ones to avoid the same issue there was with framerate consistency
 * added ticks to color speed slider and changed text under it to reflect it, it was talking about optimal values that you can't see, now it's just ticks

### CODE REVIEW

**Comment by psieg** [comment]
ok to test
Will take a look tomorrow

File: Software/src/SoundManager.hpp [hunk 1]
```diff
@@ -44,19 +43,22 @@ class SoundManager : public QObject
...
+	virtual ~SoundManager();
+	static SoundManager* create(int hWnd = 0, QObject* parent = 0);
 
 signals:
 	void updateLedsColors(const QList<QRgb> & colors);
 	void deviceList(const QList<SoundManagerDeviceInfo> & devices, int recommended);
 
 public:
-	void start(bool isMoodLampEnabled);
+	virtual void start(bool isMoodLampEnabled) { Q_UNUSED(isMoodLampEnabled); Q_ASSERT(("Not implemented", false)); };
```
Comment (psieg) [inline Software/src/SoundManager.hpp]: Please call this class SoundManagerBase. As opposed to the GrabManager it's not meant to be instantiated. And rename the bool to something like started.

File: Software/src/src.pro [hunk 2]
```diff
@@ -180,6 +182,12 @@ macx{
     HEADERS += \
     MacOSSession.h
 
+    DEFINES += SOUNDVIZ_SUPPORT
```
Comment (psieg) [inline Software/src/src.pro]: this should be in the default build_vars

File: Software/src/SoundManager.hpp (hunk 1 above)
Comment (zomfg) [reply Software/src/SoundManager.hpp]: agreed
>And rename the bool to something like started.

it was like this when I got here, I swear!

**Comment by zomfg** [comment]
> Please undo the changes to the slider. The comment about smoothness refers to the device smoothness: Quickly changing soundviz colors won't be visible if the device has a high smoothing value. Feel free to clarify that.

oooh so it's a Lightpack thing, that had me confused...
so I reverted and the label is now hidden for non-Lightpack devices

**Comment by psieg** [comment]
Yeah I didn't realize it doesn't apply to non-Lightpack. Maybe after I get around to the software smoothing. If ever...

**Pull request merged**